src/MiddlewareFactory.php

```diff
@@ -64,6 +66,14 @@ public function create(array|callable|string $middlewareDefinition): MiddlewareI
             return $this->container->get($middlewareDefinition);
         }
 
+        if ($this->isInvokableClassDefinition($middlewareDefinition)) {
+            return $this->wrapCallable([$middlewareDefinition, '__invoke']);
+        }
+
+        if ($this->isRequestHandlerClassDefinition($middlewareDefinition)) {
```

Comment:
Class that implement `RequestHandlerInterface` should have a higher priority than invokable class.